extensions/visualizations/oppia-visualization-click-hexbins.directive.spec.ts

```diff
@@ -0,0 +1,115 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive unit tests for the "click hexbins" visualization.
+ */
+
+require('visualizations/oppia-visualization-click-hexbins.directive.ts');
+
+describe('Oppia click hexbins visualization', function() {
+  let $compile, $rootScope;
+  let el: JQLite;
+
+  beforeEach(angular.mock.module('oppia', function($provide) {
+    $provide.value('AssetsBackendApiService', {
+      getImageUrlForPreview: (url: string) => url,
+    });
+    $provide.value('ContextService', {
+      getEntityId: () => 'eid',
+      getEntityType: () => 'exploration',
+    });
```

Comment:
This is perfectly fine, the other thing you could do is import TestBed and `$provide.value('ContextService', TestBed.get(ContectService));`
But what you have done is also good, in fact, I would prefer this. Maybe call this object `MockContextService` and then provide it? Your call.

Reply:
I'd prefer this since there's less indirection about what's going on, thanks though!